### PR TITLE
DM-43209: Add type annotations

### PR DESCRIPTION
**Pull request by kkeijuro**
Add type annotations to summit_extras

### CODE REVIEW

File: python/lsst/summit/utils/tmaUtils.py [hunk 1]
```diff
@@ -1326,7 +1345,7 @@ def _mergeData(self, data):
 
         return merged
 
-    def getEvent(self, dayObs, seqNum):
+    def getEvent(self, dayObs: int, seqNum: int) -> summitUtils.tmaUtils.TMAEvent | None:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1348]: Again, shouldn't need importing like this.

File: python/lsst/summit/utils/tmaUtils.py [hunk 2]
```diff
@@ -1361,7 +1380,7 @@ def getEvent(self, dayObs, seqNum):
             self.log.warning(f"Event {seqNum} not found for {dayObs}")
             return None
 
-    def getEvents(self, dayObs, addBlockInfo=True):
+    def getEvents(self, dayObs: int, addBlockInfo: bool = True) -> list[summitUtils.tmaUtils.TMAEvent]:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1383]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 3]
```diff
@@ -1482,7 +1501,9 @@ def _getEfdDataForDayObs(self, dayObs):
             merged = self._mergeData(data)
             self._data[dayObs] = merged
 
-    def _calculateEventsFromMergedData(self, data, dayObs, dataIsForCurrentDay, addBlockInfo):
+    def _calculateEventsFromMergedData(
+        self, data: pd.DataFrame, dayObs: int, dataIsForCurrentDay: bool, addBlockInfo: bool
+    ) -> list[summitUtils.tmaUtils.TMAEvent]:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1506]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 4]
```diff
@@ -1538,7 +1559,9 @@ def _calculateEventsFromMergedData(self, data, dayObs, dataIsForCurrentDay, addB
             self.addBlockDataToEvents(dayObs, events)
         return events
 
-    def _statesToEventTuples(self, states, dataIsForCurrentDay):
+    def _statesToEventTuples(
+        self, states: dict[int, summitUtils.tmaUtils.TMAEvent], dataIsForCurrentDay: bool
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1563]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 5]
```diff
@@ -1636,7 +1659,11 @@ def _statesToEventTuples(self, states, dataIsForCurrentDay):
 
         return parsedStates
 
-    def addBlockDataToEvents(self, dayObs, events):
+    def addBlockDataToEvents(
+        self,
+        dayObs: int,
+        events: list[summitUtils.tmaUtils.TMAEvent] | summitUtils.tmaUtils.TMAEvent,
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1665]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 6]
```diff
@@ -1680,7 +1707,9 @@ def addBlockDataToEvents(self, dayObs, events):
                     # dataclass attribute after creation.
                     object.__setattr__(event, "blockInfos", toSet)
 
-    def _makeEventsFromStateTuples(self, states, dayObs, data):
+    def _makeEventsFromStateTuples(
+        self, states: list[tuple[Union[Time, int, TMAState]]], dayObs: int, data: pd.DataFrame
+    ) -> list[summitUtils.tmaUtils.TMAEvent]:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1712]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 7]
```diff
@@ -1726,7 +1755,7 @@ def _makeEventsFromStateTuples(self, states, dayObs, data):
         return events
 
     @staticmethod
-    def printTmaDetailedState(tma):
+    def printTmaDetailedState(tma: summitUtils.tmaUtils.TMAStateMachine) -> None:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1758]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 8]
```diff
@@ -1789,7 +1818,12 @@ def printFullDayStateEvolution(self, dayObs, taiOrUtc="utc"):
         )
         self.printEventDetails(fakeEvent, taiOrUtc=taiOrUtc, printHeader=False)
 
-    def printEventDetails(self, event, taiOrUtc="tai", printHeader=True):
+    def printEventDetails(
+        self,
+        event: summitUtils.tmaUtils.TMAEvent,
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1823]: And here.

File: python/lsst/summit/utils/tmaUtils.py [hunk 9]
```diff
@@ -1878,7 +1912,7 @@ def printEventDetails(self, event, taiOrUtc="tai", printHeader=True):
                 tma.apply(row)
                 tmaStates[rowNum] = tma.state
 
-    def findEvent(self, time):
+    def findEvent(self, time: astropy.time.Time) -> summitUtils.tmaUtils.TMAEvent | None:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/tmaUtils.py:1915]: And here.

File: python/lsst/summit/utils/utils.py [hunk 10]
```diff
@@ -120,7 +125,7 @@ def countPixels(maskedImage, maskPlane):
     return len(np.where(np.bitwise_and(maskedImage.mask.array, bit))[0])
 
 
-def quickSmooth(data, sigma=2):
+def quickSmooth(data: np.ndarray[int], sigma: float = 2) -> np.ndarray[int]:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/utils.py:128]: I think these are almost always floats not ints in the data, both input and output.

File: python/lsst/summit/utils/utils.py [hunk 11]
```diff
@@ -291,7 +296,9 @@ def getImageStats(exp):
     return result
 
 
-def detectObjectsInExp(exp, nSigma=10, nPixMin=10, grow=0):
+def detectObjectsInExp(
+    exp: afwImage.Exposure, nSigma: int = 10, nPixMin: int = 10, grow: int = 0
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/utils.py:300]: nSigma coule be a float.

File: python/lsst/summit/utils/utils.py [hunk 12]
```diff
@@ -863,7 +883,7 @@ def starTrackerFileToExposure(filename, logger=None):
     return exp
 
 
-def obsInfoToDict(obsInfo):
+def obsInfoToDict(obsInfo: astro_metadata_translator.ObservationInfo) -> dict:
```
Comment (mfisherlevine) [inline python/lsst/summit/utils/utils.py:886]: Again, maybe just import the symbol?

File: python/lsst/summit/utils/butlerUtils.py [hunk 13]
```diff
@@ -378,7 +384,7 @@ def getExpIdFromDayObsSeqNum(butler, dataId):
     return {"exposure": expRecord.id}
 
 
-def updateDataIdOrDataCord(dataId, **updateKwargs):
+def updateDataIdOrDataCord(dataId: dict, **updateKwargs: int) -> dict:
```
[the remark this answers is not in the document]
Comment (kkeijuro) [reply python/lsst/summit/utils/butlerUtils.py:387]: It means which type are the keys, but yes, definitively not an int...

File: python/lsst/summit/utils/starTracker/starTracker.py [hunk 14]
```diff
@@ -29,6 +29,7 @@
 
 import lsst.afw.image as afwImage
 import lsst.daf.base as dafBase
+import lsst.rubintv.production as rubintvProduction
```
[the remark this answers is not in the document]
Comment (kkeijuro) [reply python/lsst/summit/utils/starTracker/starTracker.py:32]: getRawDataDirForDayObs function has a dependency on rubintv production

[the remark this answers is not in the document]
Comment (mfisherlevine) [reply python/lsst/summit/utils/starTracker/starTracker.py:32]: Well, two things:
1) it doesn't actually have a dependency on it, other than for the type
2) That's not the correct import for it - `StarTrackerCamera` is defined in this exact file!

**Pull request merged**